# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1448475255 -3600
#      Wed Nov 25 19:14:15 2015 +0100
# Node ID e6ccc8ae9e415aae0982e461925451fc8a22b161
# Parent  7d6f385419020d6f243edb5517eaeebb15db3e85
add list of mru files to the toolbar of the editor

* file-editor.cc (insert_new_open_actions): insert the new and open actions
  before the new popdown menu;
  (construct): create a tool button with the mru menu as popdown menu and
  add it to the toolbar;

* file-editor.h: new action for the new push button

diff --git a/libgui/src/m-editor/file-editor.cc b/libgui/src/m-editor/file-editor.cc
--- a/libgui/src/m-editor/file-editor.cc
+++ b/libgui/src/m-editor/file-editor.cc
@@ -1193,17 +1193,17 @@ file_editor::request_styles_preferences 
 void
 file_editor::insert_new_open_actions (QAction *new_action,
                                       QAction *new_fcn_action,
                                       QAction *open_action)
 {
   _fileMenu->insertAction (_mru_file_menu->menuAction (), open_action);
   _fileMenu->insertAction (open_action, new_fcn_action);
   _fileMenu->insertAction (new_fcn_action, new_action);
-  _tool_bar->insertAction (_save_action, open_action);
+  _tool_bar->insertAction (_popdown_mru_action, open_action);
   _tool_bar->insertAction (open_action, new_action);
 }
 
 QAction*
 file_editor::add_action (QMenu *menu, const QIcon &icon, const QString &text,
                          const char *member)
 {
   QAction *a;
@@ -1534,17 +1534,25 @@ file_editor::construct (void)
                                          SLOT (switch_right_tab ()));
   _move_tab_left_action = add_action (0, QIcon (), "",
                                       SLOT (move_tab_left ()));
   _move_tab_right_action = add_action (0, QIcon (), "",
                                        SLOT (move_tab_right ()));
 
   // toolbar
 
+  // popdown menu with mru files
+  QToolButton *popdown_button = new QToolButton ();
+  popdown_button->setToolTip (tr ("Recent Files"));
+  popdown_button->setMenu (_mru_file_menu);
+  popdown_button->setPopupMode (QToolButton::InstantPopup);
+  popdown_button->setToolButtonStyle (Qt::ToolButtonTextOnly);
+
   // new and open actions are inserted later from main window
+  _popdown_mru_action = _tool_bar->addWidget (popdown_button);
   _tool_bar->addAction (_save_action);
   _tool_bar->addAction (_save_as_action);
   _tool_bar->addSeparator ();
   _tool_bar->addAction (_print_action);
   _tool_bar->addSeparator ();
   _tool_bar->addAction (_undo_action);
   _tool_bar->addAction (_redo_action);
   _tool_bar->addAction (_copy_action);
diff --git a/libgui/src/m-editor/file-editor.h b/libgui/src/m-editor/file-editor.h
--- a/libgui/src/m-editor/file-editor.h
+++ b/libgui/src/m-editor/file-editor.h
@@ -339,16 +339,17 @@ private:
   QAction *_toggle_bookmark_action;
   QAction * _remove_bookmark_action;
 
   QAction *_print_action;
   QAction *_run_action;
   QAction *_run_selection_action;
 
   QAction *_edit_function_action;
+  QAction *_popdown_mru_action;
   QAction *_save_action;
   QAction *_save_as_action;
   QAction *_close_action;
   QAction *_close_all_action;
   QAction *_close_others_action;
 
   QAction *_redo_action;
   QAction *_undo_action;
